clients/drcachesim/tracer/instru_offline.cpp

```diff
@@ -597,8 +598,10 @@ offline_instru_t::instrument_instr(void *drcontext, void *tag, void **bb_field,
             return adjust;
         pc = dr_fragment_app_pc(tag);
     } else {
+        // This routine is called for the first instr in the expanded scatter/gather
+        // sequence, which turns out to be a non-app instr which doesn't have an app pc.
         // XXX: For repstr do we want tag insted of skipping rep prefix?
-        pc = instr_get_app_pc(app);
+        pc = scatter_gather_expanded ? dr_fragment_app_pc(tag) : instr_get_app_pc(app);
```

Comment:
The instrs added by drreg in `drx_expand_scatter_gather` are not marked as app instrs today. I suppose if we do mark them as app and set their translation, we won't need this special logic here. It seems intuitive that they should be app instrs, as they are added by the app2app phase.

But marking them as app raises another question: should reg spills and restores in the expanded sequence be counted as app memref as well?

Reply:
drreg spill instrs should not be app instrs.  We already have non-app instrs in app2app from repstr expansion and we're just living w/ the fact that clients should check for app vs tool in analysis and insertion phase.

Reply:
Dropped the idea of marking reg spill instrs as app.